src/validate.ts

```diff
@@ -288,6 +282,48 @@ function _getTargetNames(
   return names;
 }
 
+function _getUserTargets(
+  field: FieldContext,
+  ruleSchema: ValidationRuleSchema,
+  ruleName: string,
+  userMessage: string | ValidationMessageGenerator | undefined
+) {
+  const userTargets: any = {};
+  const values: Record<string, any> = field.rules[ruleName];
+  const params: RuleParamSchema[] = ruleSchema.params || [];
+
+  if (values) {
```

Comment:
The PR looks great, but I would like to simplify the flow here by eliminating some redundant nesting. From my prespective the flatter the code the easier it is to follow.

Instead of:

```js
if (values) {
 /// stuff
}
```
we can refactor to this:


```js
if (!values) {
  // early return
  return { };
}

// stuff
```

Reply:
I should have preempted that from looking at your adjustments of my previous PR.

I'll get on that :)